src/components/Facility/FacilityConfigure.tsx

```diff
@@ -1,112 +1,74 @@
-  errors: {},
-};
-
-const FormReducer = (state = initialState, action: any) => {
-  switch (action.type) {
-    case "set_form": {
-      return {
-        ...state,
-        form: action.form,
-      };
-    }
-    case "set_error": {
-      return {
-        ...state,
-        errors: action.errors,
-      };
-    }
-    default:
-      return state;
-  }
-};
+import { Submit } from "../Common/ButtonV2";
 
 interface IProps {
   facilityId: string;
 }
 
+const formSchema = z.object({
+  middleware_address: z
+    .string()
+    .nonempty({ message: "Middleware Address is required" })
+    .regex(/^(?!https?:\/\/)[a-zA-Z0-9_-]+(\.[a-zA-Z0-9_-]+)*\.[a-zA-Z]{2,}$/, {
+      message: "Invalid Middleware Address",
+    }),
+});
+
 export const FacilityConfigure = (props: IProps) => {
-  const [state, dispatch] = useReducer(FormReducer, initialState);
   const { facilityId } = props;
   const [isLoading, setIsLoading] = useState(false);
 
-  const { loading } = useTanStackQueryInstead(routes.getPermittedFacility, {
-    pathParams: { id: facilityId },
-    onResponse: (res) => {
-      if (res.data) {
-        const formData = {
-          name: res.data.name,
-          state: res.data.state,
-          district: res.data.district,
-          local_body: res.data.local_body,
-          ward: res.data.ward,
-          middleware_address: res.data.middleware_address,
-        };
-        dispatch({ type: "set_form", form: formData });
-      }
-    },
+  const { isPending: loading, data } = useQuery({
+    queryKey: [routes.getPermittedFacility.path, facilityId],
+    queryFn: query(routes.getPermittedFacility, {
+      pathParams: { id: facilityId },
+    }),
   });
 
-  const handleSubmit = async (e: React.FormEvent) => {
-    e.preventDefault();
-    setIsLoading(true);
-    if (!state.form.middleware_address) {
-      dispatch({
-        type: "set_error",
-        errors: { middleware_address: ["Middleware Address is required"] },
-      });
-      setIsLoading(false);
-      return;
-    }
-    if (
-      state.form.middleware_address.match(
-        /^(?!https?:\/\/)[a-zA-Z0-9_-]+(\.[a-zA-Z0-9_-]+)*\.[a-zA-Z]{2,}$/,
-      ) === null
-    ) {
-      dispatch({
-        type: "set_error",
-        errors: {
-          middleware_address: ["Invalid Middleware Address"],
-        },
-      });
-      setIsLoading(false);
-      return;
-    }
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (!data) return;
 
-    const data = {
-      ...state.form,
-      middleware_address: state.form.middleware_address,
+    const formData = {
+      name: data.name,
+      state: data.state,
+      district: data.district,
+      local_body: data.local_body,
+      ward: data.ward,
+      middleware_address: values.middleware_address,
     };
 
+    setIsLoading(true);
+
     const { res, error } = await request(routes.partialUpdateFacility, {
       pathParams: { id: facilityId },
-      body: data,
+      body: formData,
```

Comment:
Go ahead and switch this to using `useMutate` (similar to useQuery above).

Reply:
done